apps/token-manager/app/src/components/LocalIdentitiesAutoComplete/LocalIdentitiesAutoComplete.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import { useAragonApi } from '@aragon/api-react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { EthIdenticon, IdentityBadge, theme } from '@aragon/ui'
+import AutoCompleteSelected from '../AutoComplete/AutoCompleteSelected'
+
+const withKey = item => ({ key: item.address, ...item })
+const sortAlphAsc = (a, b) => a.name.localeCompare(b.name)
+
+const LocalIdentitiesAutoComplete = React.memo(
+  React.forwardRef(function LocalIdentitiesAutoComplete(
+    { onChange, wide, value, required },
+    ref
+  ) {
+    const { api } = useAragonApi()
+    const [items, setItems] = useState([])
+    const [selected, setSelected] = useState(null)
+    const [searchTerm, setSearchTerm] = useState('')
+
+    const handleSearch = useCallback(
+      async term => {
+        if (term.length < 3) {
+          setItems([])
+          return
+        }
+        const items = await api.searchIdentities(term).toPromise()
+        setItems(items.map(withKey).sort(sortAlphAsc))
+      },
+      [api]
+    )
+    const handleChange = useCallback(
+      value => {
+        setSearchTerm(value)
+        handleSearch(value)
+        onChange(value)
+      },
+      [onChange]
+    )
+    const handleSelect = useCallback(
+      selected => {
+        const { name, address } = selected
+        setSearchTerm(name)
+        handleSearch(name)
```

Comment:
Why do we search here upon selection?

Reply:
If the user clicks on the selected item the input box receives focus and the list of options is rendered, without the list might show more items than it should (number of items when they clicked on the item)